crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -898,6 +915,18 @@ impl RoomInfo {
         self.base_info.create.as_ref()?.as_original()?.content.room_type.as_ref()
     }
 
+    /// Replace our latest_event with the supplied event, and delete it and all
+    /// older encrypted events from latest_encrypted_events, given that the
+    /// new event was at the supplied index in the latest_encrypted_events
+    /// list.
+    pub fn latest_event_decrypted(&mut self, latest_event: SyncTimelineEvent, index: usize) {
+        self.latest_event = Some(latest_event);
+        for _ in 0..index {
+            // TODO: do this more efficiently, maybe through a drop_n method on RingBuffer?
```

Comment:
Ideally, we would want to use `VecDeque::drain`, that's probably the easiest and the most efficient.

```rust
self.latest_encrypted_events.drain(0..=index)
```

or something like that.

Reply:
See https://github.com/matrix-org/matrix-rust-sdk/pull/2183 for `RingBuffer::drain`.

Reply:
Done, thanks!